src/app/home/home.component.html

```diff
@@ -46,7 +46,7 @@
 
           <mat-paginator
             [pageIndex]="page"
-            [length]="items.length"
+            [length]="filteredLength(items)"
```

Comment:
minor: it's always better to avoid function call in the template and use signals or static values